tests/smoke_tests/run_smoke_test.py

```diff
@@ -69,6 +75,23 @@ async def run_smoke_test(
                 dataset_path="examples/datasets/cifar_data/",
                 checkpoint_path="examples/assets/best_checkpoint_fczjmljm.pkl",
                 assert_evaluation_logs=True,
+                seed=42,
+                server_metrics={
+                    "rounds": {
+                        "1": {
+                            "metrics_aggregated": {"val - prediction - accuracy": 0.4744},
+                            "loss_aggregated": (2.001, 0.05), # to override default tolerance, pass it in as a tuple
```

Comment:
Rather than using a tuple here, which might be a little hard to understand without the comment you put here (i.e. without always replicating that comment), maybe we can just use a nested dictionary? That is
```
"loss_aggregated": {"target_value": 2.001, "custom_tolerance": 0.05},
```

Then in the parsing of `_assert_metrics_dict`, we could do
```
tolerance = DEFAULT_TOLERANCE
if isinstance(value, dict):
    if "custom_tolerance" in value:
        tolerance = value["custom_tolerance"]
    value = value["target_value"]
```
Maybe it belabors the parsing a bit. Just trying to think of a way to make the custom tolerance override a bit more "obvious."

Reply:
What about a class? It would make things even more organized. I thought about it but went with the tuple, but I think it makes sense to do it.
```python
class MetricValue:
    def __init__(self.value, self.tolerance=DEFAULT_TOLERANCE):
        self.value = value
        self.tolerance = tolerance
```
```python
"loss_aggregated": MetricValue(2.001, 0.05),
```

Reply:
oh well, just doing the json file change and it doesn't make sense to make it a class. Will do the dictionary change.